hr_holidays_leave_request_wizard/wizards/hr_leave_wizard.py

```diff
@@ -0,0 +1,553 @@
+            day.name = babel.dates.format_skeleton(
+                skeleton='MMMEd',
+                datetime=datetime.combine(day.date, time.min),
+                locale=locale,
+            )
+
+    @api.multi
+    def _generate_intervals(self):
+        HrLeaveWizardDayInterval = self.env['hr.leave.wizard.day.interval']
+        for day in self:
+            employee = day.wizard_id.employee_id
+            if not employee.resource_id:  # pragma: no cover
+                day.interval_ids = [(5, 0, 0)]
+                continue
+
+            day_start = datetime.combine(day.date, time.min)
+            day_end = datetime.combine(day.date, time.max)
+            day_start = day_start.replace(tzinfo=utc)
+            day_end = day_end.replace(tzinfo=utc)
+
+            calendar = employee.resource_calendar_id
+            attendance_intervals = calendar._attendance_intervals(
+                day_start,
+                day_end,
+                employee.resource_id,
+            )
+            unpaid_intervals = calendar._leave_intervals(
+                day_start,
+                day_end,
+                employee.resource_id,
+            )
+            global_intervals = calendar._leave_intervals(
+                day_start,
+                day_end,
+                None,
+            )
+            intervals = (
+                attendance_intervals - (unpaid_intervals - global_intervals)
+            )
+
+            interval_ids = [(5, 0, 0)]
+            for start, stop, meta in intervals:
+                values = HrLeaveWizardDayInterval._prepare_values(
+                    start.astimezone(utc).replace(tzinfo=None),
+                    stop.astimezone(utc).replace(tzinfo=None),
+                )
+                interval_ids += [(0, 0, values)]
+            day.interval_ids = interval_ids
+
+    @api.multi
+    def _recompute_request_amount_upward(self):
+        for day in self:
+            day.request_amount = sum(
+                day.interval_ids.mapped('request_amount')
+            ) / len(day.interval_ids) if day.interval_ids else 0.0
+
+    @api.multi
+    def _recompute_request_amount_downward(self):
+        for day in self:
+            if day.requested_hours >= day.available_hours \
+                    and day.available_hours > 0.0:
+                for interval in day.interval_ids:
+                    interval.request_amount = 1.0
+            elif day.requested_hours <= 0.0 or day.available_hours <= 0.0:
+                for interval in day.interval_ids:
+                    interval.request_amount = 0.0
+            else:
+                requested_hours = day.requested_hours
+                for interval in day.interval_ids:
+                    if interval.available_hours > 0:
+                        interval.request_amount = min(
+                            requested_hours,
+                            interval.available_hours
+                        ) / interval.available_hours
```

Comment:
Maybe `ZeroDivisionError` ? 
(and also few other places where` / `occurs , Just a comment on first look)

Reply:
`if interval.available_hours > 0:` ?